[docs] Add documentation on iOS credentials (#6329)
expo/expo: docs/pages/versions/unversioned/distribution/app-signing.md

title: App signing
---

Expo automates the process of signing your app for iOS and Android, but in both cases you can choose to provide your own overrides. Both `expo build:ios` and
`expo build:android` commands generate signed applications ready to be uploaded into respective stores.

On this page, we'll talk about the credentials that each platform requires. If you're curious about how we store your credentials on our end, take a look at our [security documentation](../security/).

## iOS

The 3 primary iOS credentials, all of which are associated with your Apple Developer account, are:

- Distribution Certificate
- Provisioning Profiles
- Push Notification Keys

> Whether you let Expo handle all your credentials, or you handle them all yourself, it's smart to understand what each of these credentials mean, when and where they're used, and what happens when they expire or are revoked.

Reviewer: ```suggestion
Whether you let Expo handle all your credentials, or you handle them all yourself, it can be valuable to understand what each of these credentials mean, when and where they're used, and what happens when they expire or are revoked.
```

### Distribution Certificate

> Whereas the rest of these credentials are about the apps themselves, the certificate is all about you, the developer. You may only have one distribution certificate associated with your Apple Developer account.

Reviewer: ```suggestion
The distribution certificate is all about you, the developer, and not about any particular app. You may only have one distribution certificate associated with your Apple Developer account.
```

> This certificate will be used for all of your apps. If this certificate expires, your apps in production will not be affected. However, you will need to generate a new certificate if you want to upload new apps to the app store, or update any of your existing apps. Deleting a distribution certificate has no effect on any apps already on the App Store.

Reviewer: ```suggestion
This certificate will be used for all of your apps. If this certificate expires, your apps in production will not be affected. However, you will need to generate a new certificate if you want to upload new apps to the App Store, or update any of your existing apps. Deleting a distribution certificate has no effect on any apps already on the App Store.
```

### Push Notification Keys

> Apple Push Notification Keys, or APN keys, allow the associated apps to send and receive push notifications. By default, any app built with Expo will require an APN key. This is so that you can enable push notifications for your app through a quick [OTA update](../../guides/configuring-ota-updates/), rather than needing to submit an entirely new binary.

Reviewer: ```suggestion
Apple Push Notification Keys (often abbreviated as APN keys) allow the associated apps to send and receive push notifications. By default, any app built with Expo will require an APN key. This is so that you can enable push notifications for your app through a quick [OTA update](../../guides/configuring-ota-updates/), rather than needing to submit an entirely new binary.
```

> You can have a maximum of 2 APN keys associated with your Apple Developer account. However, multiple different apps can use the same APN key. If you revoke an APN key, all apps that rely on that key will no longer be able to send or recieve push notifications until you upload a new key to replace it. Push notification keys do not expire.

Reviewer: ```suggestion
You can have a maximum of 2 APN keys associated with your Apple Developer account, and a single key can be used with any number of apps. If you revoke an APN key, all apps that rely on that key will no longer be able to send or receive push notifications until you upload a new key to replace it. Push notification keys do not expire.
```
